src/components/Pagination/Pagination.test.tsx

```diff
@@ -0,0 +1,42 @@
+import { render, screen } from '@testing-library/react';
+import { describe, it, expect } from 'vitest';
+import type { ReactElement } from 'react';
+import Pagination from './Pagination';
+import { MemoryRouter } from 'react-router-dom';
+
+function renderWithRouter(ui: ReactElement, route = '/?page=2') {
+  window.history.pushState({}, 'Test page', route);
+  return render(<MemoryRouter>{ui}</MemoryRouter>);
+}
+
+describe('Pagination Component', () => {
+  it('renders pagination with multiple pages', () => {
+    renderWithRouter(<Pagination current={2} total={5} />);
+
+    expect(screen.getByText('1')).toBeVisible();
+    expect(screen.getByText('2')).toBeVisible();
+    expect(screen.getByText('3')).toBeVisible();
+    expect(screen.getByText('4')).toBeVisible();
+    expect(screen.getByText('5')).toBeVisible();
+    expect(screen.getByText('→')).toBeVisible();
+    expect(screen.getByText('←')).toBeVisible();
+  });
```

Comment:
```suggestion
  it.each(['1', '2', '3', '4', '5', '→', '←'])(
    'render pagination item %s',
    (i) => expect(screen.getByText(i)).toBeVisible(),
  );
```
you can create tests on the fly from arrays